### ISSUE TITLE
Licensing for fruit power up graphics and stone fruit graphics and Johan’s songs

### ISSUE THREAD

**Issue by datahead8888** [Secretchronicles/TSC#356]
No license is listed for Bugsbane's fruit power up grahics.  We probably need to change it to "special", based on our prior discussions.

By extension, the license on the stone fruit graphics probably needs to be changed to "special".

The stone army shell graphic can probably stay as GPLv3+, since I think the army shell graphics came from the secretmaryo.org forums.  It would not be a bad idea to double check this.

Lastly, we need to check any other graphics by Bugsbane to see if any similar licensing issues exist, and we would probably want to explicitly mark the license on these.

This is a legal issue, so I am marking it as release-2.0.0 and as Blocker.

**Comment by datahead8888** [comment]
This issue is *very* big.

I checked settings files for these items.  They were uploaded by Bugsbane to git, and we are not currently aware of any other license statement.  They probably default to the exclusive special license:
* The title screen logo does not have a license specified
* The question block replacements do not have a license specified
* The spin block replacements do not have a license specified
* The ground_3 image has no license specified
* The coin hud image has no license specified
* The jewel graphics do not have a license specified
* The power up replacements (fruit) do not have a license specified
    * By extension, my stone fruit graphics *might* inherit any such problems (from the SVG license)

These are the items by Johan Brodd that fall under special:
* The title theme
* The credits theme

As Quintus has pointed out, special licenses can be a major issue when submitting to Linux distributions or if others fork the game.  We are looking at contacting Bugsbane and Johan about these issues.

Related (long but informative) discussion:
http://lists.secretchronicles.de/tsc-devel/2015/02/0000003.html

**Comment by Bugsbane** [comment]
Augh. Sorry, guys. Had some major life stuff come up, which is why I haven't been around. Just tell me what you'd like my graphics licensed as, and I'm happy to provide a statement that they're licensed to that effect. I'm happy with any OSI / FSF approved license.

**Comment by datahead8888** [comment]
Bugsbane, it's great to see you again!  :)  I can't type up a good response now, since I've got to go teach, but I just wanted to say that, and I hope all is well for you.

**Comment by datahead8888** [comment]
Bugsbane, I'd like to let Quintus speak as to what's needed on licensing, to help make sure we request exactly what's needed.

Quintus, don't forget to explain what's needed for the SVG licenses.  I in turn can make sure the derivative stone fruit SVG's I made are properly licensed.

**Comment by Quintus** [comment]
@Bugsbane, very, very good to see you back. Hope you got well through your “life stuff” :-)

> Just tell me what you'd like my graphics licensed as, and I'm happy to provide a statement that they're licensed to that effect. I'm happy with any OSI / FSF approved license.

I would have suggested [CC-BY 4.0](https://creativecommons.org/licenses/by/4.0/) as the license, because CC licenses fit better on graphical assets than the typical sourcecode licenses, but if you would rather like to use a OSI/FSF-approved license I suggest the [BSD 2-clause license](http://opensource.org/licenses/BSD-2-Clause).

As for the requirements, there have been some commits with your graphics. Since we have been properly using either the `--author` option of `git commit` or you did the commits yourself, determining which commits need confirmation is easy (`git log --author=Bugsbane --oneline`):

* 6d14e38 Commiting Bugsbane's SVG's for fruit power ups
* d81fb7a Update ground_3 graphics and HUD Coins image
* a45dd0a Replace SMB coins with gems graphics + settings files
* 7be5424 Non SMB power up items graphics and settings
* 9595df4 Update yellow boxes (except power block) and settings files
* f01687a added source files for gems, green_3 and spin crate
* c94f028 update tiles resolution and author settings (again)
* 5d04092 increased resolution of spin crate frames
* 7d9a37e Update author for green_3, gems and spin crates
* 3b7fa8c turned spin boxes into wooden crates
* eaa6575 fix final visual issues w green_3 ground
* 8b07431 update settings files to make gems smaller
* f927c34 increased resolution of gems and small visual fixes
* 9366f5e changed coins to gems
* 3bd238b fixed green_3 tiling
* 5b6f4c2 refine green_3 ground tiles further

Additionally, I added the logo before we started using `--author` properly:

* 23ef1e1 Add temporary new logo. Cf. ticket #203.

So, could you please state here in the tracker that the graphics that have been introduced with the commits in the above lists are licensed by you under the terms of either the CC-BY 4.0 or the BSD 2-clause license? (or any other license you find suitable of course)

--------

When this is done there’s a big stone off my heart. If you read the discussion on the mailinglist, there have been the weirdest ideas popping up on how to continue...

On a side note, to prevent such things in the future, I’m working on some “constitutional” document that outlines the modalities of working on the game.

Vale,
Quintus

**Comment by Bugsbane** [comment]
The graphics that have been introduced in the commits above are being made available by me, the creator of said graphics, to this project under the Creative Commons CC-BY 4.0 license, today the 11'th of February, 2015.

As to the settings files, what's the standard code license for the project? GPL v3? Anyway, let me know and I'll do a similar release for those, too.

**Comment by Quintus** [comment]
Thank you so much!

> As to the settings files, what's the standard code license for the project? GPL v3? Anyway, let me know and I'll do a similar release for those, too.

I don’t think the settings files are even copyrightable at all because they are too small entities. Four lines of plaintext usually don’t constitute something you can copyright except you made a poem or so. If you wanted to license them, I’d suggest to just use CC-BY 4.0 as well, because they’re so closely related to the graphics that licensing them differently does not make sense in my opinion.

Vale,
Quintus

**Comment by datahead8888** [comment]
This ticket should stay open until we finish checking with Johan on his music and change the license for the stone fruit graphics (I will try and do this when I get some time).  We also might consider whether the stone army shell graphic can be put under CC-BY 4.0 or whether it has to stay under the GPL based license I put it under.

**Comment by Quintus** [comment]
> This ticket should stay open until we finish checking with Johan on his music

I’m fine with that.

> We also might consider whether the stone army shell graphic can be put under CC-BY 4.0 or whether it has to stay under the GPL based license I put it under.

If anyone is able to explain the confusing wording of the Old SMC Contribution license to me I can tell you that. Put it under the GPLv3+, and the answer whether the OSCL requires you to do so becomes meaningless. That’s safest.

Vale,
Quintus

**Comment by datahead8888** [comment]
> I’m fine with that.

Thanks, Quintus.  I at least wanted to be able to track the follow up with Johan for release 2.0.  I can try to look at the licenses on the stone fruit graphics when I get some time.

**Comment by Quintus** [comment]
Note I have added the CC-BY 4.0 marks to Bugsbane’s graphics today, so you don’t have to do that.

Vale,
Quintus

**Comment by datahead8888** [comment]
No, it's the two stone fruit settings files in https://github.com/Secretchronicles/TSC/blob/release-2.0.0/tsc/data/pixmaps/ground/castle_1/statue/objects/berry.settings.  They still need to be updated.  I've made a license statement in the stone graphics ticket.  I'd be glad to change these settings files when I get time again.

**Comment by datahead8888** [comment]
No, I'm not trying to be a pain, but this is still a Blocker task, unfortunately...

**Comment by Quintus** [comment]
>  If this must be done, my suggestion is to either use songs with permissive licenses or one of our existing songs for now.

I think we could just return to the `hyper_1` music temporarily we alraedy again that was used before Johan submitted his songs.

Vale,
Quintus

**Comment by datahead8888** [comment]
Quintus reviewed my email.  I have now emailed Johan at both his gmail email account and his soundcloud account.

**Comment by sydneyjd** [comment]
@datahead8888 @Quintus Still no word from Johan? 
Ugg, i'd hate to have to pull all the work he did back out, but if we must, we must. :disappointed:

**Comment by datahead8888** [comment]
> Ugg, i'd hate to have to pull all the work he did back out, but if we must, we must.

Yeah, this is pretty bad.  We still have no confirmation.  I felt the same way about removing the turtle boss in release 2.0, but that is going to happen, too.  The good news is that these assets still can be replaced in later 2.X and 3.X releases.

**Comment by Quintus** [comment]
I have removed Johan’s songs for now and replaced by `hyper_1`. This can be changed back anytime when we get a license confirmation.

Vale,
Quintus

**Comment by Quintus** [comment]
Note: I’d rather not want to close this ticket so it stays open as a reminder we need to get after the license confirmation sooner or later. Thus I just removed it from the 2.0.0 milestone.

Vale,
Quintus

**Comment by datahead8888** [comment]
I could tell Johan in another message that we've removed it for now and want to add it back, once we get a license, if this message works well.

Also, we might consider making a licensing or legal task.  It seems to keep showing up in one form or another.

**Comment by Quintus** [comment]
> Also, we might consider making a licensing or legal label. 

Done.

Vale,
Quintus

**Comment by datahead8888** [comment]
We never got a license confirmation on the two songs (title and credits).  They were then removed from the game.  All the other stuff involving fruit graphics licensing is done.  Do we want to keep this ticket open?

If we want a task for the title / credits themes (probably to be obtained through the contest), it would probably be clearer just to make a new task for that, since this was a legal licensing task.

**Comment by Quintus** [comment]
I agree. We can close this.

Valete,
Quintus
